File: ui_examples/views/MarkdownExample/app/src/main/java/lee/dorian/android/markdown_ex/MainActivity.java
package lee.dorian.android.markdown_ex;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.widget.EditText;

import us.feras.mdv.MarkdownView;

public class MainActivity extends AppCompatActivity {

    private EditText etContent;
    private MarkdownView mvContent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etContent = (EditText)findViewById(R.id.etContent);
        mvContent = (MarkdownView)findViewById(R.id.mvContent);
        etContent.addTextChangedListener(etContentTextWatcher);
    }

    private TextWatcher etContentTextWatcher = new TextWatcher() {

        @Override
        public void beforeTextChanged(CharSequence s, int start, int count, int after) {
        }

        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {
        }

        @Override
        public void afterTextChanged(Editable s) {
            mvContent.loadMarkdown(etContent.getText().toString());
        }

    };

}

File: ui_examples/views/MarkdownExample/app_kotlin/src/main/java/lee/dorian/android/markdown_ex/kotlin/MainActivity.kt
package lee.dorian.android.markdown_ex.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.EditText
import us.feras.mdv.MarkdownView

class MainActivity : AppCompatActivity() {

    private lateinit var etContent: EditText
    private lateinit var mvContent: MarkdownView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        etContent = findViewById(R.id.etContent) as EditText
        mvContent = findViewById(R.id.mvContent) as MarkdownView
        etContent.addTextChangedListener(etContentTextWatcher)
    }

    private val etContentTextWatcher: TextWatcher = object : TextWatcher {

        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
        }

        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
        }

        override fun afterTextChanged(s: Editable?) {
            mvContent.loadMarkdown(etContent.text.toString())
        }

    }

}